### PR TITLE
docs: migrate to nuxt

### PR DESCRIPTION
**Pull request by mattmaribojoc**
# Related issue

Closes #

# Scope of work

- uses the new docs theme to build the docs
- adds custom logic to allow for the code injection from the source code
- adds links to qwik documentation
- migrates all of the pages
     - there are some exceptions to this because the "common pages" like theming, customization were merged into one page with the code snippets changing depending on which framework was selected
- temporarily skip the cypress tests on the docs (talked to @FRSgit) to unblock the upgrade to newer node versions

# Screenshots of visual changes

image

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [ ] cypress tests created

### CODE REVIEW

File: apps/docs/components/ai-assistant.md
Comment (FRSgit) [inline apps/docs/components/ai-assistant.md:1]: should this me removed? 😄

**Review by Szymon-dziewonski — COMMENTED** [review]
code looking good, however I couldnt run it locally, got many errors
```
szymons-mbp-1:components szymondziewonski$ yarn dev:docs
Nuxt 3.6.5 with Nitro 2.6.3                                                                                                             12:51:55 PM
                                                                                                                                        12:51:56 PM
  ➜ Local:    http://localhost:3000/
  ➜ Network:  use --host to expose

ℹ Using default Tailwind CSS file                                                                                     nuxt:tailwindcss 12:51:57 PM

[12:51:57 PM]  ERROR  [unhandledRejection] Cannot find package 'tailwind-config-viewer' imported from /Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/tailwindcss/dist/module.mjs

  at __node_internal_captureLargerStackTrace (node:internal/errors:496:5)
  at new NodeError (node:internal/errors:405:5)
  at packageResolve (node:internal/modules/esm/resolve:890:9)
  at moduleResolve (node:internal/modules/esm/resolve:939:20)
  at defaultResolve (node:internal/modules/esm/resolve:1132:11)
  at nextResolve (node:internal/modules/esm/loader:163:28)
  at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
  at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
  at ESMLoader.import (node:internal/modules/esm/loader:524:22)
  at importModuleDynamically (node:internal/modules/esm/translators:110:35)
  at importModuleDynamicallyCallback (node:internal/process/esm_loader:36:14)
  at setupViewer (/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/tailwindcss/dist/module.mjs:186:24)
  at setup (/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/tailwindcss/dist/module.mjs:301:9)
  at async normalizedModule (/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxt/kit/dist/index.mjs:2141:17)
  at async installModule (/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxt/kit/dist/index.mjs:2439:95)
  at async initNuxt (/Users/szymondziewonski/Projects/storefront-ui/node_modules/nuxt/dist/index.mjs:3238:7)
  at async NuxtDevServer._load (/Users/szymondziewonski/Projects/storefront-ui/node_modules/nuxi/dist/chunks/dev2.mjs:255:5)
  at async NuxtDevServer.load (/Users/szymondziewonski/Projects/storefront-ui/node_modules/nuxi/dist/chunks/dev2.mjs:187:7)
  at async NuxtDevServer.init (/Users/szymondziewonski/Projects/storefront-ui/node_modules/nuxi/dist/chunks/dev2.mjs:182:5)
  at async Object.run (/Users/szymondziewonski/Projects/storefront-ui/node_modules/nuxi/dist/chunks/dev-child.mjs:103:5)
  at async runCommand$1 (/Users/szymondziewonski/Projects/storefront-ui/node_modules/nuxi/dist/shared/nuxi.9d864d83.mjs:1647:7)
  at async runCommand$1 (/Users/szymondziewonski/Projects/storefront-ui/node_modules/nuxi/dist/shared/nuxi.9d864d83.mjs:1638:11)
  at async runMain$1 (/Users/szymondziewonski/Projects/storefront-ui/node_modules/nuxi/dist/shared/nuxi.9d864d83.mjs:1770:7)

                                                                                                                                        12:51:57 PM
SEO Kit v1.3.9 • All-in-one SEO by @harlan_zw                                                                                           12:51:57 PM
  └─ 💖 Like this package? Consider sponsoring me on GitHub https://github.com/sponsors/harlan-zw                                       12:51:57 PM
                                                                                                                                        12:51:57 PM
✔ Nuxt DevTools is enabled v0.8.3 (experimental)                                                                                       12:51:59 PM
ℹ [content-assets] Websocket listening on "ws://localhost:4001/"                                                                       12:51:59 PM

[12:51:59 PM]  WARN  Duplicated imports "useAlgoliaFacetedSearch", the one from "/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/algolia/dist/runtime/composables/useAlgoliaFacetedSearch.d.ts" has been ignored


[12:51:59 PM]  WARN  Duplicated imports "useAlgoliaInitIndex", the one from "/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/algolia/dist/runtime/composables/useAlgoliaInitIndex.d.ts" has been ignored


[12:51:59 PM]  WARN  Duplicated imports "useAlgoliaRecommend", the one from "/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/algolia/dist/runtime/composables/useAlgoliaRecommend.d.ts" has been ignored


[12:51:59 PM]  WARN  Duplicated imports "useAlgoliaRef", the one from "/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/algolia/dist/runtime/composables/useAlgoliaRef.d.ts" has been ignored


[12:51:59 PM]  WARN  Duplicated imports "useAlgoliaSearch", the one from "/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/algolia/dist/runtime/composables/useAlgoliaSearch.d.ts" has been ignored


[12:51:59 PM]  WARN  Duplicated imports "useAsyncAlgoliaSearch", the one from "/Users/szymondziewonski/Projects/storefront-ui/node_modules/@nuxtjs/algolia/dist/runtime/composables/useAsyncAlgoliaSearch.d.ts" has been ignored


 ERROR  Cannot start nuxt:  Cannot find module 'postcss-custom-properties'                                                              12:52:00 PM
Require stack:
```

Thats probably the same issue with your build, IMHO docs cannot be part of workspace, since `yarn v1` hoist all packages so packages clash with `preview/nuxt` and either docs or preview have no idea which package version pick

**Comment by filrak** [comment]
@mattmaribojoc do you think we can finish that and merge to keep consistency in the docs?

File: apps/docs/components/ai-assistant.md
Comment (mattmaribojoc) [reply apps/docs/components/ai-assistant.md:1]: yup...it will be moved to the main docs.

**Review by Szymon-dziewonski — COMMENTED** [review]
Hey, great effort done here! 


I have couple of questions:
- we do not have in docs anymore this tip tag (e.g Badge component in Source tab)
image
Is it intentional to remove this info?

- white mode missing border in showcases
image

- is it a thing that source has scrollbar in wierd spot? not at the bottom of container?
image

- IMHO showcases are too big currently, documentation pages are twice long in compare to docs on vuepress
image

In old docs
image

And for example in banners, footer - height of showcases are too small, I think those iframes should be customized and set to desired height :)

- Something happened to Prose styling page http://localhost:3000/customization/prose#example-content-styled-with-prose, content inside is smaller then iframe
image

**Comment by mattmaribojoc** [comment]
Thanks for the good review @Szymon-dziewonski  :)

- Respect the passed `min-height` on Showcases:
image 
 - also fixes `Prose` Showcase 
image
- Upstream fix for Code Block Scrollbar 
image
- Add reusable partial content block for creating the callout
image
- Border on Light Mode Showcases
image

File: apps/docs/components/content/_components/button.md [hunk 1]
```diff
@@ -74,84 +75,92 @@ You can truncate button content with Tailwind using `max-w-` and `truncate` clas
...
+::react-only
+<<<../../../../preview/next/pages/showcases/Button/ButtonContent.tsx
+::
+
 </Showcase>
 
 ## Accessibility notes
 
-Button component can be rendered as `<button>` or `<a>` or any other tag by providing it with prop <!-- vue -->`tag`<!-- end vue --><!-- react -->`as`<!-- end react -->. When rendered as `<button>` this component applies default `type='button'` if no other `type` is passed as a prop.
+Button component can be rendered as `<button>` or `<a>` or any other tag by providing it with prop ::vue-only`tag`::::react-only`as`::. When rendered as `<button>` this component applies default `type='button'` if no other `type` is passed as a prop.
```
[the remark this answers is not in the document]
Comment (mattmaribojoc) [reply apps/docs/components/content/_components/button.md]: image

File: apps/docs/components/content/_components/chip.md [hunk 2]
```diff
@@ -20,27 +20,29 @@ Internally, `SfChip` uses a hidden `<input type="checkbox">` element to handle t
...
+<<<../../../../preview/next/pages/showcases/Chip/ChipSizes.tsx
+::
 
 </Showcase>
 
 ### Thumbnails
 
 <!-- TODO: Add linking to thumbnail component when ready -->
-`SfChip` provides <!-- vue -->`prefix` and `suffix` slots<!-- end vue --><!-- react -->`slotPrefix` and `slotSuffix` props<!-- end react --> that you can use to add custom content before/after your default content. This can be useful for adding things like icons, badges, or `SfThumbnail` to your toggleable chip.
+`SfChip` provides ::vue-only`prefix` and `suffix` slots::::react-only`slotPrefix` and `slotSuffix` props:: that you can use to add custom content before/after your default content. This can be useful for adding things like icons, badges, or `SfThumbnail` to your toggleable chip.
```
[the remark this answers is not in the document]
Comment (mattmaribojoc) [reply apps/docs/components/content/_components/chip.md]: image

File: apps/docs/components/content/_components/input.md [hunk 3]
```diff
@@ -93,17 +96,18 @@ This is an example of what `SfInput` might look like in your end code. It has a
...
+::react-only
+<<<../../../../preview/next/pages/showcases/Input/InputCharacters.tsx
+::
+
 </Showcase>
 
 ## Notes
 
-All non-prop attributes and styles added to `SfInput` component are passed directly to the native input element. This means that you can add all of the input attributes directly to `SfInput`. If you want to style the wrapper `div`, you can pass your classes via the <!-- vue -->`wrapperClass`<!-- end vue --><!-- react --> `wrapperClassName`<!-- end react -->prop. 
+All non-prop attributes and styles added to `SfInput` component are passed directly to the native input element. This means that you can add all of the input attributes directly to `SfInput`. If you want to style the wrapper `div`, you can pass your classes via the ::vue-only`wrapperClass`::::react-only `wrapperClassName`::prop.
```
[the remark this answers is not in the document]
Comment (mattmaribojoc) [reply apps/docs/components/content/_components/input.md]: image

File: apps/docs/components/content/_components/select.md [hunk 4]
```diff
@@ -20,27 +22,27 @@ If you need to make this field required, it is crucial to communicate this inten
...
+::react-only
+<<<../../../../preview/next/pages/showcases/Select/SelectSizes.tsx
+::
 
 </Showcase>
 
 ### Custom chevron
 
-You can replace the default chevron with your own custom content by using the <!-- vue -->`chevron` slot<!-- end vue --> <!-- react -->`slotChevron` prop<!-- end react -->.
+You can replace the default chevron with your own custom content by using the ::vue-only`chevron` slot:: ::react-only`slotChevron` prop::.
```
[the remark this answers is not in the document]
Comment (mattmaribojoc) [reply apps/docs/components/content/_components/select.md]: image

File: apps/docs/components/content/_components/thumbnail.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+::
+::react-only
+<<<../../../../preview/next/pages/showcases/Thumbnail/Gradient.tsx
+::
+
+</Showcase>
+
+### Image
+
+You can pass an image ::vue-onlyinto the default slot:: ::react-onlyas children:: to achieve avatar-like look of thumbnail.
```
[the remark this answers is not in the document]
Comment (mattmaribojoc) [reply apps/docs/components/content/_components/thumbnail.md]: image

**Comment by Szymon-dziewonski** [comment]
@mattmaribojoc looking great, double checked docs after your fixes, have no issues, huge thanks for this PR 🙇

**Review by Szymon-dziewonski — APPROVED** [review]

**Review by FRSgit — APPROVED** [review]

**Pull request merged**